src/jit/codegenxarch.cpp

```diff
@@ -2637,9 +2637,18 @@ void CodeGen::genLclHeap(GenTreePtr tree)
         inst_IV(INS_push_hide, 0); // --- push 8-byte 0
         inst_IV(INS_push_hide, 0); // --- push 8-byte 0
 #elif defined(_TARGET_X86_)
+#if !defined(FEATURE_PAL)
         // Push a single 4-byte zero. This matches the 4-byte STACK_ALIGN value.
         static_assert_no_msg(STACK_ALIGN == REGSIZE_BYTES);
         inst_IV(INS_push_hide, 0); // --- push 4-byte 0
+#else
+        // Push four 4-byte zeros. This matches the 16-byte STACK_ALIGN value.
+        static_assert_no_msg(STACK_ALIGN == REGSIZE_BYTES * 4);
```

Comment:
I think that it is better to replace all of these #ifdefs with a loop that pushes STACK_ALIGN bytes of zeros onto the stack.

```
static_assert_no_msg((STACK_ALIGN % REGSIZE_BYTES) == 0));
unsigned const count = (STACK_ALIGN / REGSIZE_BYTES);

for (unsigned i=0; i<count; i++)
{
 inst_IV(INS_push_hide, 0); // --- push REG_SIZE bytes of 0 
}
// Note that the stack must always be aligned to STACK_ALIGN bytes
```